components/MyCourses/MyCourses.js

```diff
@@ -20,18 +17,22 @@ import LocalizationProvider from '@mui/lab/LocalizationProvider';
 import DateTimePicker from '@mui/lab/DateTimePicker';
 import EditIcon from '@mui/icons-material/Edit';
 import { useState } from 'react';
+import Link from 'next/link'
+
 import MainButton from '../basecomponents/MainButton.js'
+import defaultProfilePic from '../../utils/constants/index.js'
 
-export default function MyCourses({ course }) {
+export default function MyCourses({ course, handleDeleteCourse, index }) {
   const [showStudentList, setShowStudentList] = useState(false);
   const [editCourseInfo, setEditCourseInfo] = useState(false);
   const [userType, setUserType] = useState('mentor');
   const [currCourseName, setCurrCourseName] = useState(course.courseName);
   const [currStartTime, setCurrStartTime] = useState(course.startTime);
   const [currEndTime, setCurrEndTime] = useState(course.endTime);
   const [newCourseName, setNewCourseName] = useState('');
-  const [newStartTime, setNewStartTime] = useState(null);
-  const [newEndTime, setNewEndTime] = useState(null);
+  const [newStartTime, setNewStartTime] = useState(currStartTime);
+  const [newEndTime, setNewEndTime] = useState(currEndTime);
+  const [currentIndex, setCurrentIndex] = useState(index);
```

Comment:
Add to tech debt. Refactor to useReducer after midpoint demo. Hook useReducer is built on top of useState to handle complex state logic like we have above.